superset/assets/visualizations/deckgl/layers/scatter.jsx

```diff
@@ -68,8 +89,10 @@ function getLayer(formData, payload, slice, inFrame) {
     data = jsFnMutator(data);
   }
 
-  if (inFrame != null) {
-    data = data.filter(inFrame);
+  if (filters != null) {
+    filters.forEach((f) => {
+      data = data.filter(f);
```

Comment:
interesting pattern here :)

Reply:
Functional JS! ;)

Maybe we should call this `postFilters` or something similar, since it's a series of filters applied after the query (due to clicks from the legend or the play slider).